include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -739,29 +697,40 @@ class FEPointEvaluation
   /**
    * The desired update flags for the evaluation.
    */
-  UpdateFlags update_flags;
+  const UpdateFlags update_flags;
 
   /**
-   * The update flags specific for the mapping in the fast evaluation path.
+   * The FEValues object underlying the slow evaluation path.
    */
-  UpdateFlags update_flags_mapping;
+  std::shared_ptr<FEValues<dim, spacedim>> fe_values;
 
   /**
-   * The FEValues object underlying the slow evaluation path.
+   * Pointer to mapping info on the fly computed during reinit.
    */
-  std::shared_ptr<FEValues<dim, spacedim>> fe_values;
+  std::unique_ptr<NonMatching::MappingInfo<dim, spacedim>>
+    mapping_info_on_the_fly;
 
   /**
-   * Array to store temporary data computed by the mapping for the fast
-   * evaluation path.
+   * Pointer to currently used mapping info (either on the fly or external
+   * precomputed).
    */
-  internal::FEValuesImplementation::MappingRelatedData<dim, spacedim>
-    mapping_data;
+  NonMatching::MappingInfo<dim, spacedim> *mapping_info;
 
   /**
    * The reference points specified at reinit().
    */
   std::vector<Point<dim>> unit_points;
+
+  /**
+   * Bool indicating if the constructor with precomputed mapping info has been
+   * called.
+   */
+  const bool precomputed_mapping;
```

Comment:
Do we really need this? I would have thought that we can query it from the `mapping_info_on_the_fly` object, which is the `nullptr` if we construct with an external `MappingInfo`, but non-zero if we construct it inside this class.

Reply:
I removed the `precomputed_mapping` bool and now query the information from `mapping_info_on_the_fly`.